deno/lib/helpers/util.ts

```diff
@@ -25,28 +25,35 @@ export namespace util {
   };
 
   export const getValidEnumValues = (obj: any) => {
-    const validKeys = Object.keys(obj).filter(
+    const validKeys = objectKeys(obj).filter(
       (k: any) => typeof obj[obj[k]] !== "number"
     );
     const filtered: any = {};
     for (const k of validKeys) {
       filtered[k] = obj[k];
     }
-    return getValues(filtered);
-  };
-
-  export const getValues = (obj: any) => {
-    return Object.keys(obj).map(function (e) {
-      return obj[e];
-    });
+    return objectValues(filtered);
   };
 
   export const objectValues = (obj: any) => {
-    return Object.keys(obj).map(function (e) {
+    return objectKeys(obj).map(function (e) {
       return obj[e];
     });
   };
 
+  export const objectKeys: ObjectConstructor["keys"] =
+    typeof Object.keys === "function" // eslint-disable-line ban/ban
+      ? (obj: any) => Object.keys(obj) // eslint-disable-line ban/ban
+      : (object: any) => {
+          const keys = [];
+          for (const key in object) {
+            if (Object.prototype.hasOwnProperty.call(object, key)) {
+              keys.push(key);
+            }
+          }
+          return keys;
+        };
```

Comment:
I'd be more comfortable with e.g. es-shims' implementation, unless I misunderstand how Deno works.

Reply:
what do you mean? 
I have no experience with deno, but looks like ponyfill is not needed there because Object.keys should be defined

Reply:
If we know that a ponyfill is not needed then why are we checking if `Object.keys` is defined? (I have no experience with Deno either.)

Reply:
@wilmveel or @jeremyBanks do you have thoughts on shimming in Deno? Is it necessary or not?

Reply:
> E: wait, is this generated code from the "normal" version? In that case disregard this comment.

yep